File: src/main/java/com/alxnns1/mobhunter/entities/renderer/GoldenfishRenderer.java
package com.alxnns1.mobhunter.entities.renderer;

import com.alxnns1.mobhunter.MobHunter;
import com.alxnns1.mobhunter.entities.GoldenfishEntity;
import com.alxnns1.mobhunter.entities.model.GoldenfishModel;
import net.minecraft.client.renderer.entity.EntityRendererManager;
import net.minecraft.client.renderer.entity.MobRenderer;
import net.minecraft.util.ResourceLocation;
import net.minecraftforge.api.distmarker.Dist;
import net.minecraftforge.api.distmarker.OnlyIn;

import javax.annotation.Nonnull;

@OnlyIn(Dist.CLIENT)
public class GoldenfishRenderer extends MobRenderer<GoldenfishEntity, GoldenfishModel<GoldenfishEntity>> {
	public GoldenfishRenderer(EntityRendererManager renderManagerIn) {
		super(renderManagerIn, new GoldenfishModel<>(), 0.5f);
	}

	@Nonnull
	@Override
	public ResourceLocation getEntityTexture(@Nonnull GoldenfishEntity entity) {
		return new ResourceLocation(MobHunter.MOD_ID, "textures/entity/goldenfish.png");
	}
}

File: src/main/kotlin/com/alxnns1/mobhunter/entity/fish/renderer/GoldenfishRenderer.kt
package com.alxnns1.mobhunter.entity.fish.renderer

import com.alxnns1.mobhunter.MobHunter
import com.alxnns1.mobhunter.entity.MHRenderer
import com.alxnns1.mobhunter.entity.fish.GoldenfishEntity
import com.alxnns1.mobhunter.entity.fish.model.GoldenfishModel
import net.minecraft.client.renderer.entity.EntityRendererManager
import net.minecraft.util.ResourceLocation

private val RL = ResourceLocation(MobHunter.MOD_ID, "textures/entity/goldenfish.png")

class GoldenfishRenderer(rendererManager: EntityRendererManager) : MHRenderer<GoldenfishEntity, GoldenfishModel>(rendererManager, GoldenfishModel(), 0.5F) {

	override fun getEntityTexture(entity: GoldenfishEntity): ResourceLocation = RL
}
